src/main/java/com/aol/cyclops/types/IterableFunctor.java

```diff
@@ -44,21 +46,34 @@ A potentially asynchronous merge operation where data from each publisher may ar
      * 
      * 
      */
-    default  ReactiveSeq<T> mergePublisher(Collection<? extends Publisher<T>> publishers, QueueFactory<T> factory){
+    default  ReactiveSeq<T> mergePublisher(Iterable<? extends Publisher<T>> publishers, QueueFactory<T> factory){
         Counter c = new Counter();
-        c.active.set(publishers.size()+1);
-        QueueBasedSubscriber<T> init = QueueBasedSubscriber.subscriber(factory,c,publishers.size());
+        
+        int size = 0;
+        Iterator<? extends Publisher<T>> it = publishers.iterator();
```

Comment:
I think that traverse once iterables such as  jooλ Seq, or ReactiveSeq would blow up on subsequent use (after size is determined).

The most efficient algorithm may be to check if publishers implements Collection, if not perform a collect operation on it (ReactiveSeq.fromIterable(publishers).toList()) and then continue as previous impl.

What do you think?